Source/WebCore/editing/Editor.cpp

```diff
@@ -4594,8 +4594,8 @@ FontAttributes Editor::fontAttributesAtSelectionStart()
     if (foregroundColor.isValid() && !Color::isBlackColor(foregroundColor))
         attributes.foregroundColor = foregroundColor;
 
-    if (auto* shadowData = style->textShadow())
-        attributes.fontShadow = { style->colorWithColorFilter(shadowData->color()), { shadowData->x().value, shadowData->y().value }, shadowData->radius().value };
+    if (!style->textShadow().isEmpty())
+        attributes.fontShadow = { style->colorWithColorFilter(style->textShadow()[0].color), { style->textShadow()[0].location.x().value, style->textShadow()[0].location.y().value }, style->textShadow()[0].blur.value };
```

Comment:
Is there a way to make this code look less repetitive without an object lifetime problem?
```suggestion
    if (auto& shadow = style->textShadow(); !shadow.isEmpty())
        attributes.fontShadow = { style->colorWithColorFilter(shadow[0].color), { shadow[0].location.x().value, shadow[0].location.y().value }, shadow[0].blur.value };
```
Would be nice to avoid the repeated `[0]` also, but I couldn’t figure out how to do that without making it even wordier.

Or maybe we can make this share a little more with `EllipsisBoxPainter`. Not critical I guess.